### ISSUE TITLE
ReceiveFrameString() within Task (Thread) won't cancel/close

### ISSUE THREAD

**Issue by Sn3b** [zeromq/netmq#417]
Hi,

I'm having a little problem with ReceiveFrameString() that's running within a Task I'm trying to cancel.

I know ReceiveFrameString() blocks its current thread until a new message arrives, but it prevents me from cancelling the Task it's running in.

Should I use another method? Or is there a more clever way to do it?

Thanks! :)

**Comment by Sn3b** [comment]
I use TryReceiveFrameString() with a timeout instead, so it doesn't block forever, and I can cancel my task (abort my thread).

Seems to work better :)

**Comment by somdoron** [comment]
The best way to do it with a Poller, you poll for receive ready (you get an
event when message is ready) and then receive the message, it is guaranteed
not to block. You can cancel the poller at anytime. Read the docs about
poller.

**Comment by Sn3b** [comment]
Wow! Very cool thanks!

My previous solution worked but using the poller is a lot more elegant!

I had to adopt quite a few changes due to architecture restrictions but I really like this new solution. Thanks to the poller I was able to get rid of this ugly while(true), and thanks PollTillCancelledNonBlocking I don't even need to create a new thread/task and bother with cancellation token etc...

Thanks a lot somdoron, and thank you so much for all the doc, good articles etc... Loving NetMQ since I discovered it!
